src/urllib3/http2.py

```diff
@@ -163,19 +235,60 @@ def getresponse(  # type: ignore[override]
             data=bytes(data),
         )
 
+    def request(  # type: ignore[override]
+        self,
+        method: str,
+        url: str,
+        body: _TYPE_BODY | None = None,
+        headers: typing.Mapping[str, str] | None = None,
+        *,
+        preload_content: bool = True,
+        decode_content: bool = True,
+        enforce_content_length: bool = True,
+        **kwargs: typing.Any,
+    ) -> None:
+        """Send an HTTP/2 request"""
+        if "chunked" in kwargs:
+            # TODO this is often present from upstream.
+            # raise NotImplementedError("`chunked` isn't supported with HTTP/2")
+            pass
+
+        if self.sock is not None:
+            self.sock.settimeout(self.timeout)
+
+        self.putrequest(method, url)
+
+        headers = headers or {}
+        for k, v in headers.items():
+            self.putheader(k, v)
+
+        if not self._has_header(b"user-agent"):
```

Comment:
Since we know `_headers` is all lowercase bytes headers, can we do `b"user-agent" not in self._headers`?